### PR TITLE
fix(FileListener): Listen to file movements across share boundaries

### LINKED ISSUE #1118: Recognize results are lost when transferring files to other user
### Which version of recognize are you using?

6.1.1

### Enabled Modes

Object recognition, Face recognition

### TensorFlow mode

Normal mode

### Downstream App

Files App

### Which Nextcloud version do you have installed?

28.0.4

### Which Operating system do you have installed?

Debian 12

### Which database are you running Nextcloud on?

MariaDB 10.11.6

### Which Docker container are you using to run Nextcloud? (if applicable)

_No response_

### How much RAM does your server have?

16GB

### What processor Architecture does your CPU have?

x86_64

### Describe the Bug

When files are transferred from one user to another user via transfer ownership, only the classification results are transferred, face detections are lost. Also, the transferred files are not added to the face detection queue for detecting the faces again.

### Expected Behavior

Either keep the face recognition and run the clustering for the new user or re-run both face recognition and clustering after moving the files.

### To Reproduce

- Upload images to one user
- let recognize do its thing
- transfer ownership of these images to another user
- see that face detections are lost and files are not added to face detection queue

### Debug log

_No response_

### PR DESCRIPTION
**Pull request by marcelklehr**
fixes #1118

### CODE REVIEW

File: .github/workflows/lint-php-cs.yml [hunk 1]
```diff
@@ -43,7 +43,9 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Install dependencies
-        run: composer i
+        run: |
+          composer i
+          composer i
```
Comment (julien-nc) [inline .github/workflows/lint-php-cs.yml:48]: Any reason to run it twice?

File: lib/Hooks/FileListener.php [hunk 2]
```diff
@@ -315,6 +333,43 @@ public function postInsert(Node $node, bool $recurse = true): void {
...
+		/** @var array{users:array<string,array{node_id:int, node_path: string}>, remote: array<string,array{node_id:int, node_path: string}>, mail: array<string,array{node_id:int, node_path: string}>} $targetAccessList */
+		$targetAccessList = $this->shareManager->getAccessList($target, true, true);
+		$targetUserIds = array_map(fn ($id) => strval($id), array_keys($targetAccessList['users']));
+
+		$usersToAdd = array_diff($targetUserIds, $this->sourceUserIds);
+		$existingUsers = array_diff($targetUserIds, $usersToAdd);
+		// *handwaving* I know this is a stretch but it's good enough
+		$ownerId = $existingUsers[0];
+
+		if ($target->getType() === FileInfo::TYPE_FOLDER) {
```
Comment (julien-nc) [inline lib/Hooks/FileListener.php:346]:
```suggestion
		if ($target instanceof Folder) {
```

**Review by julien-nc — APPROVED** [review]
LGTM. I'm not familiar enough with the code to make a full in-depth review but the tests cover this change so :+1:

File: .github/workflows/lint-php-cs.yml (hunk 1 above)
Comment (marcelklehr) [reply .github/workflows/lint-php-cs.yml:48]: Yeah. because of using php scoper the composer binary deps are not installed properly the first time :D

**Pull request merged**